contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/HadoopStoreBuilder.java

```diff
@@ -409,6 +380,69 @@ public void build() {
     }
 
     /**
+     * This function configures a {@link JobConf} instance based on the {@param size} of the data
+     * that Voldemort will need to serve. If the size provided is that of the input data, then
+     * these settings will be calculated in a best-effort basis. It is not possible to predict in
+     * advance the correct settings based on the input data, since compression factors may vary
+     * a lot between data sets.
+     *
+     * On the other hand, if the size is that of the mappers' output, then it should be possible
+     * to predict proper settings. There is still another edge case, however, which we cannot
+     * avoid completely: if the data is heavily skewed towards a hot partition, then it may not
+     * be possible at all to tweak these settings correctly.
+     *
+     * @param size of the dataset
+     * @param conf MapReduce {@link JobConf} for which to set properties.
+     * @return
+     */
+    private void setNumChunksAndNumReduceTasks(long size, JobConf conf) {
```

Comment:
Since we are not retrying the job automatically in this modified PR, unlike my original PR, I don't think it's worth having this logic in a separate function. It makes the flow more confusing without actually reducing code size, since the function is now called only once. Furthermore, the java doc is a bit misleading, since there is now never any case where we call this function with the mapper's output as the `size` parameter.

For these reasons, I'd recommend leaving this code where it was originally, rather than moving it into a function.